Sprint-2/implement/tally.js

```diff
@@ -1,3 +1,17 @@
-function tally() {}
-
+function tally(list) {
+    let count = {};
+    if(list.length == 0){
+        return {};
+    }
+    if(!Array.isArray(list)){
+        throw new Error("invalid input, input must be an array");
+    }
+    else{
```

Comment:
Because `throw` and `return` both immediately stop running your function, you can actually get rid of the else here - that can be useful for stopping your code always being more and more indented, and to be able to more clearly read the flow of the function. You can write this:
```js
if (!Array.isArray(list)) {
    throw new Error("...");
}
for (let item of list) {
  ...
}
return count;
```

The code you wrote is not wrong - it absolutely works fine, just this form may be a little more clear.